primitives/src/communities.rs

```diff
@@ -30,7 +32,46 @@ pub type CommunityIndexType = u32;
 pub type LocationIndexType = u32;
 pub type Degree = I64F64;
 pub type NominalIncome = I64F64;
-pub type CommunityIdentifier = H256;
+
+#[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, Default, RuntimeDebug, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct CommunityIdentifier {
+    location: [u8; 5],
+    bootstrappers: u32,
+}
+
+impl CommunityIdentifier {
+    pub fn new<AccountId: Encode>(location: Location, bootstrappers: Vec<AccountId>) -> CommunityIdentifier {
+        let geo_hash = GeoHash::try_from_params(location.lat, location.lon, 5).unwrap();
+        let crc_engine = Crc::<u32>::new(&CRC_32_CKSUM);
+        let mut digest = crc_engine.digest();
+        digest.update(&(bootstrappers).encode());
+        CommunityIdentifier {
+            location: [
+                geo_hash[0],
+                geo_hash[1],
+                geo_hash[2],
+                geo_hash[3],
+                geo_hash[4],
+            ],
+            bootstrappers: digest.finalize(),
+        }
+    }
+
+    pub fn as_array(self) -> [u8; 9] {
+        [
+            self.location[0],
+            self.location[1],
+            self.location[2],
+            self.location[3],
+            self.location[4],
+            ((self.bootstrappers >> 24) & 0xff) as u8,
```

Comment:
this doesn't not look very idiomatic. after the above, this will be easy to simplify

Reply:
I don't know if the new version is better.. while it does not access the array contents in a hardcoded fashion, it does copy around a lot. I was unable to find a non-copying version of merging two fixed sized arrays of different size that does not require unsafe code

Reply:
Use: https://github.com/canndrew/concat-arrays